elements/pl-external-grader-variables/pl-external-grader-variables.py

```diff
@@ -1,34 +1,90 @@
 import chevron
-import lxml.html
+import lxml
 import prairielearn as pl
 
-PARAMS_NAME_DEFAULT = None
+EMPTY_DEFAULT = False
 
 
-def prepare(element_html, element_index, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    pl.check_attribs(element, ["params-name"], [])
+    pl.check_attribs(element, ["params-name"], ["empty"])
 
-    params_name = pl.get_string_attrib(element, "params-name", PARAMS_NAME_DEFAULT)
-    if params_name is None:
-        raise Exception('Attribute "params-name" is not defined.')
-    if params_name not in data["params"]:
-        raise Exception(
-            f"Variable name {params_name} does not exist in data['params']."
-        )
+    params_name = pl.get_string_attrib(element, "params-name")
 
+    # Get frontend variables
+    html_variables = []
 
-def render(element_html, element_index, data):
+    for child in element:
+        if child.tag == "pl-variable":
+            pl.check_attribs(child, ["name", "type"], [])
+
+            var_dict = {
+                "name": pl.get_string_attrib(child, "name"),
+                "type": pl.get_string_attrib(child, "type"),
+            }
+
+            description_html = pl.inner_html(child)
+
+            # Check for empty string, inner_html doesn't return None
+            if description_html:
+                var_dict["description"] = description_html
+
+            html_variables.append(var_dict)
```

Comment:
Nit: I believe this will be equivalent, and is a little terser?

```suggestion
            var_dict = {
                "name": pl.get_string_attrib(child, "name"),
                "type": pl.get_string_attrib(child, "type"),
                # Handle empty string, inner_html doesn't return None
                "description": pl.inner_html(child) or None,
            }

            html_variables.append(var_dict)
```